network-libp2p/src/discovery/peer_contacts.rs

```diff
@@ -299,35 +316,43 @@ impl PeerContactBook {
     /// If the filter matches the services provided by the contact, it is added.
     /// Otherwise it is ignored.
     /// The services_filter argument to this function contains the services that are required.
-    pub fn insert_filtered(&mut self, contact: SignedPeerContact, services_filter: Services) {
+    pub fn insert_filtered(
+        &mut self,
+        contact: SignedPeerContact,
+        services_filter: Services,
+        only_secure_connections: bool,
+    ) {
         let info = PeerContactInfo::from(contact);
 
-        if self
+        // A peer is interested to us in two cases:
+        // - I'm configured as a validator, and the peer is also a validator, then that peer is interesting to me
+        //   regardless of the services that are provided by that peer.
+        // - There is a match in the services provided by the peer and our services
```

Comment:
```suggestion
        // - The services provided by the peer are a superset of the requested services.
```
It was unclear to me what a "match" was supposed to mean. Digging into the source code, it seems to mean "superset".